File: python/app/service.py
class DirectoryService:
    def __init__(self, client):
        self._client = client

    def getPhoneNumbers(self, country: str, phone: str) -> str:
        payload = self._client.fetch_country(country)
        records = payload.get("data") or []
        if not records:
            return "-1"

        calling_codes = records[0].get("callingCodes") or []
        if not calling_codes:
            return "-1"

        calling_code = calling_codes[-1]
        return "+" + calling_code + " " + phone

File: javascript/src/service.js
function createDirectoryService(client) {
  return {
    async getPhoneNumbers(country, phone) {
      const payload = await client.fetchCountry(country);
      const records = Array.isArray(payload && payload.data) ? payload.data : [];
      if (records.length === 0) {
        return '-1';
      }

      const callingCodes = Array.isArray(records[0].callingCodes) ? records[0].callingCodes : [];
      if (callingCodes.length === 0) {
        return '-1';
      }

      const callingCode = callingCodes[callingCodes.length - 1];
      return '+' + callingCode + ' ' + phone;
    },
  };
}

module.exports = { createDirectoryService };
